src/libs/ajax/Cbas.js

```diff
@@ -1,15 +1,66 @@
 import _ from 'lodash/fp';
-import { authOpts, fetchCbas, jsonBody } from 'src/libs/ajax/ajax-common';
+import qs from 'qs';
+import { authOpts, fetchCbas, fetchCromwell, fetchLeo, fetchOk, jsonBody } from 'src/libs/ajax/ajax-common';
+import { getConfig } from 'src/libs/config';
 
 export const Cbas = (signal) => ({
   status: async (cbasUrlRoot) => {
-    const res = await fetchCbas(cbasUrlRoot)('status', _.mergeAll([authOpts(), { signal, method: 'GET' }]));
+    const res = await fetchOk(`${cbasUrlRoot}/status`, _.mergeAll([authOpts(), { signal, method: 'GET' }]));
     return res.json();
   },
   methods: {
     post: async (cbasUrlRoot, payload) => {
-      const res = await fetchCbas(cbasUrlRoot)('api/batch/v1/methods', _.mergeAll([authOpts(), jsonBody(payload), { signal, method: 'POST' }]));
+      const res = await fetchCbas(cbasUrlRoot)('methods', _.mergeAll([authOpts(), jsonBody(payload), { signal, method: 'POST' }]));
       return res.json();
     },
+    getWithoutVersions: async (cbasUrlRoot) => {
+      const keyParams = qs.stringify({ show_versions: false });
+      const res = await fetchCbas(cbasUrlRoot)(`methods?${keyParams}`, _.mergeAll([authOpts(), { signal, method: 'GET' }]));
+      return res.json();
+    },
+  },
+});
+
+export const Cromwell = (signal) => ({
+  workflows: (workflowId) => {
+    return {
+      metadata: async (includeKey, excludeKey) => {
+        const keyParams = qs.stringify({ includeKey, excludeKey }, { arrayFormat: 'repeat' });
+        const res = await fetchCromwell(`${workflowId}/metadata?${keyParams}`, { signal, method: 'GET' });
+        return res.json();
+      },
+    };
+  },
+});
+
+export const WorkflowScript = (signal) => ({
+  get: async (workflowUrl) => {
+    const res = await fetchOk(workflowUrl, { signal, method: 'GET' });
+    return res.text();
+  },
+});
+
+export const extractLeoTokenFromCookies = (cookieString) => {
+  const cookies = cookieString.split(';');
+  return _.flow(
+    _.map((c) => c.trim()),
+    _.filter((c) => c.startsWith('LeoToken=')),
+    _.map((c) => c.substring(9)),
+    _.first
+  )(cookies);
+};
+
+const leoToken = () => {
+  const cookieString = document.cookie;
+  return extractLeoTokenFromCookies(cookieString);
+};
+const authHeader = { headers: { Authorization: `Bearer ${leoToken()}` } };
```

Comment:
This is definitely worth looking into - since it is working for now, would it make sense to separate this into another ticket? I am worried that this will be a Russian nesting doll issue and would rather not grow the scope of this ticket.